main.go

```diff
@@ -31,50 +29,33 @@ type Attachment struct {
 	FooterIcon *string  `json:"footer_icon"`
 }
 
+type Payload struct {
+	Parse       string       `json:"parse,omitempty"`
+	Username    string       `json:"username,omitempty"`
+	IconUrl     string       `json:"icon_url,omitempty"`
+	IconEmoji   string       `json:"icon_emoji,omitempty"`
+	Channel     string       `json:"channel,omitempty"`
+	Text        string       `json:"text,omitempty"`
+	Attachments []Attachment `json:"attachments,omitempty"`
+}
+
 func (attachment *Attachment) AddField(field Field) *Attachment {
 	attachment.Fields = append(attachment.Fields, &field)
 	return attachment
 }
 
-func Payload(text, username, imageOrIcon, channel string, attachments []Attachment) map[string]interface{} {
-	payload := make(map[string]interface{})
-	payload["parse"] = "full"
-	if username != "" {
-		payload["username"] = username
-	}
-
-	if strings.HasPrefix("http", imageOrIcon) {
-		payload["icon_url"] = imageOrIcon
-	} else if imageOrIcon != "" {
-		payload["icon_emoji"] = imageOrIcon
-	}
-
-	if channel != "" {
-		payload["channel"] = channel
-	}
-
-	if text != "" {
-		payload["text"] = text
-	}
-
-	if len(attachments) > 0 {
-		payload["attachments"] = attachments
-	}
-
-	return payload
-}
-
 func redirectPolicyFunc(req gorequest.Request, via []gorequest.Request) error {
 	return fmt.Errorf("Incorrect token (redirection)")
 }
 
-func Send(webhookUrl string, proxy string, payload map[string]interface{}) []error {
-	data, _ := json.Marshal(payload)
+func Send(token string, proxy string, payload Payload) []error {
+	webhookUrl := fmt.Sprintf("https://hooks.slack.com/services/%v", token)
```

Comment:
@adrianlzt Just happen to see this, can we please stick with full webhook url instead of just token? That way we can support Slack API Compatible services (like [Rocket.Chat](https://github.com/RocketChat/Rocket.Chat/issues/202), etc.) as well.